packages/dd-trace/src/appsec/iast/analyzers/nosql-injection-mongodb-analyzer.js

```diff
@@ -103,7 +106,11 @@ class NosqlInjectionMongodbAnalyzer extends InjectionAnalyzer {
     })
   }
 
-  _isVulnerableRange (range) {
+  _isVulnerableRange (range, value) {
+    const rangeIsWholeValue = range.start === 0 && range.end === value?.length
```

Comment:
I think it's more safe to check `range` first or:

```suggestion
    const rangeIsWholeValue = range?.start === 0 && range?.end === value?.length
```

Reply:
true, as it was checked before, we should continue checking.